docs/high-avaliability/prometheus-HA.md

```diff
@@ -1,12 +1,62 @@
+# Running Promscale itself in High-Availability mode
+
+Promscale is a stateless service, thus it can safely be run with multiple
+replicas. A load-balancer can simply route Promscale requests to any replica.
+
 # Using Promscale with Prometheus in High Availability
 
-Promscale supports running Prometheus in High Availability (HA). This means that more than one Prometheus instances can run in parallel. Promscale implements leader election to support Prometheus HA mode through pg_advisory_lock.
+Promscale supports running Prometheus in High Availability (HA). This means
+that more than you can have multiple Prometheus instances running as "HA
+pairs". Such "HA pairs" (what we'll call clusters from now on) are simply two
+or more identical Prometheus instance running on different
+machines/containers. These instances scrape the same targets and should thus
+have the same data (modulo timing of the scrapes).
+
+Promscale supports de-duplicating data coming from such HA clusters upon
+ingest. It does this by electing a leader Prometheus instance and only
+ingesting data from that leader. If the leader Prometheus stops sending data
+to Promscale for any reason, Promscale will fail-over to the other Prometheus
+instance.
+
+Promscale supports having multiple clusters sending data to the same set of
+Promscale instances. It will elect one leader per cluster.
```

Comment:
For more clear understanding, we can add (after the end of first sentence) `You can think of a **cluster** as a **group** of similar Prometheus instances that are scraping same targets and **replicas** as individual Prometheus instance in that group, each scraping the same targets.`

Reply:
Added that to the first paragraph, not here